test-suite/src/function/custom.rs

```diff
@@ -0,0 +1,103 @@
+use {
+    crate::*,
+    gluesql_core::{
+        data::ValueError,
+        executor::{AlterError, EvaluateError},
+        prelude::{Payload, PayloadVariable, Value::*},
+        translate::TranslateError,
+    },
+};
+
+test_case!(custom, async move {
+    let test_cases = [
+        ("CREATE FUNCTION add_none ()", Ok(Payload::Create)),
+        (
+            "CREATE FUNCTION add_none ()",
+            Err(AlterError::FunctionAlreadyExists("add_none".to_owned()).into()),
+        ),
+        (
+            "CREATE FUNCTION add_one (n INT, x INT DEFAULT 1) RETURN n + x",
+            Ok(Payload::Create),
+        ),
+        (
+            "CREATE FUNCTION add_two (n INT, x INT DEFAULT 1, y INT) RETURN n + x + y",
```

Comment:
Here, if the arg value is `NULL` then it looks that the default expr is used.
But when we use `value.validate_type`, `NULL` is always valid because all the params in SQL are nullable.

How about allowing `DEFAULT` for the last arguments.

e.g.
```sql
CREATE FUNCTION add_one (n INT, x INT DEFAULT 1) RETURN n + x
```
Then, we can call `add_one` like..
```sql
add_one(1);
add_one(1, 2);
```

and in the case of two defaults,
```sql
CREATE FUNCTION add_two (n INT, x INT DEFAULT 1, y INT DEFAULT 23) RETURN n + x + y
```
Then,
```sql
add_two(1);
add_two(1, 2);
add_two(1, 2, 3);
```
Users can omit the last arguments if they want.

---
and 
```sql
CREATE FUNCTION add_two (n INT, x INT DEFAULT 1, y INT) RETURN n + x + y
```
the above would be invalid because there's no `DEFAULT` expr for the last param `y INT`.